File: ProjectWorkMilanoFE/src/app/model/JobOffer.ts
import { CompanyClient } from "./CompanyClient";
import { JobOfferSkill } from "./JobOfferSkill";

export class JobOffer {

    /*Attributi*/
    private _idJobOffer!: number;
    private _description!: String;
    private _endDate!: Date;
    private _maxRal!: number;
    private _minRal!: number;
    private _startDate!: Date;
    private _title!: String;

    /*Collegamento a ID Company Client*/
    private _companyClient!: CompanyClient;

    /*Liste*/
    private _jobOfferSkill!: JobOfferSkill[];

    /*I GETTER e i SETTER ci servono? Non mi sembra*/
    public get idJobOffer(): number {
        return this._idJobOffer;
    }
    public set idJobOffer(value: number) {
        this._idJobOffer = value;
    }

    public get description(): String {
        return this._description;
    }
    public set description(value: String) {
        this._description = value;
    }

    public get endDate(): Date {
        return this._endDate;
    }
    public set endDate(value: Date) {
        this._endDate = value;
    }

    public get maxRal(): number {
        return this._maxRal;
    }
    public set maxRal(value: number) {
        this._maxRal = value;
    }

    public get minRal(): number {
        return this._minRal;
    }
    public set minRal(value: number) {
        this._minRal = value;
    }

    public get startDate(): Date {
        return this._startDate;
    }
    public set startDate(value: Date) {
        this._startDate = value;
    }

    public get title(): String {
        return this._title;
    }
    public set title(value: String) {
        this._title = value;
    }

    public get companyClient(): CompanyClient {
        return this._companyClient;
    }
    public set companyClient(value: CompanyClient) {
        this._companyClient = value;
    }

    public get jobOfferSkill(): JobOfferSkill[] {
        return this._jobOfferSkill;
    }
    public set jobOfferSkill(value: JobOfferSkill[]) {
        this._jobOfferSkill = value;
    }
}

File: ProjectWorkMilanoBE/src/main/java/com/model/JobOffer.java
package com.model;

import java.io.Serializable;
import java.sql.Date;
import java.util.List;

import com.fasterxml.jackson.annotation.JsonIgnore;

import jakarta.persistence.CascadeType;
import jakarta.persistence.Entity;
import jakarta.persistence.FetchType;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.ManyToOne;
import jakarta.persistence.NamedQuery;
import jakarta.persistence.OneToMany;
import jakarta.persistence.Table;
import jakarta.persistence.Temporal;
import jakarta.persistence.TemporalType;

/**
 * The persistent class for the job_offer database table.
 * 
 */
@Entity
@Table(name = "job_offer")
@NamedQuery(name = "JobOffer.findAll", query = "SELECT j FROM JobOffer j")
public class JobOffer implements Serializable {
	private static final long serialVersionUID = 1L;

	@Id
	@GeneratedValue(strategy = GenerationType.IDENTITY)
	private int idJobOffer;

	private String description;

	@Temporal(TemporalType.DATE)
	private Date endDate;

	private int maxRal;

	private int minRal;

	@Temporal(TemporalType.DATE)
	private Date startDate;

	private String title;

	// bi-directional many-to-one association to CompanyClient
	@JsonIgnore
	@ManyToOne	
	@JoinColumn(name = "idCompanyClient")
	private CompanyClient companyClient;

	// bi-directional many-to-one association to ContractType
	@JsonIgnore
	@ManyToOne	
	@JoinColumn(name = "idContractType")
	private ContractType contractType;

	// bi-directional many-to-one association to JobOfferSkill
	@JsonIgnore

	@OneToMany(mappedBy = "jobOffer", fetch =FetchType.EAGER,cascade= CascadeType.REMOVE)	
	
	private List<JobOfferSkill> jobOfferSkill;

	public JobOffer() {
	}

	public int getIdJobOffer() {
		return this.idJobOffer;
	}

	public void setIdJobOffer(int idJobOffer) {
		this.idJobOffer = idJobOffer;
	}

	public String getDescription() {
		return this.description;
	}

	public void setDescription(String description) {
		this.description = description;
	}

	public Date getEndDate() {
		return this.endDate;
	}

	public void setEndDate(Date endDate) {
		this.endDate = endDate;
	}

	public int getMaxRal() {
		return this.maxRal;
	}

	public void setMaxRal(int maxRal) {
		this.maxRal = maxRal;
	}

	public int getMinRal() {
		return this.minRal;
	}

	public void setMinRal(int minRal) {
		this.minRal = minRal;
	}

	public Date getStartDate() {
		return this.startDate;
	}

	public void setStartDate(Date startDate) {
		this.startDate = startDate;
	}

	public String getTitle() {
		return this.title;
	}

	public void setTitle(String title) {
		this.title = title;
	}

	public CompanyClient getCompanyClient() {
		return this.companyClient;
	}

	public void setCompanyClient(CompanyClient companyClient) {
		this.companyClient = companyClient;
	}

	public ContractType getContractType() {
		return this.contractType;
	}

	public void setContractType(ContractType contractType) {
		this.contractType = contractType;
	}

	public List<JobOfferSkill> getJobOfferSkills() {
		return this.jobOfferSkill;
	}

	public void setJobOfferSkill(List<JobOfferSkill> jobOfferSkill) {
		this.jobOfferSkill = jobOfferSkill;
	}

}